dask_expr/groupby.py

```diff
@@ -285,9 +275,27 @@ def __init__(
         sort=None,
         observed=None,
         dropna=None,
+        slice=None,
     ):
+        by_ = by if isinstance(by, (tuple, list)) else [by]
+        self._slice = slice
+        # Check if we can project columns
+        projection = None
+        if (
+            np.isscalar(slice)
+            or isinstance(slice, (str, list, tuple))
+            or (
+                (is_index_like(slice) or is_series_like(slice))
+                and not is_dask_collection(slice)
```

Comment:
What are the constraints on `slice` here? Is it even possible for `slice` to be a Dask collection?

Reply:
It could be a Dask Index for example, so yes this should be possible. I took the logic from dask/dask though.